x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java

```diff
@@ -65,9 +65,14 @@ protected void doExecute(Task task, Request request, ActionListener<Response> li
                         model.infer(stringObjectMap, request.getUpdate(), chainedTask)));
 
                 typedChainTaskExecutor.execute(ActionListener.wrap(
```

Comment:
I wish our `ActionListener` classes had a `finally` clause....Not to signal the listener but to do things like our release or close here.